mlflow/tracking/client.py

```diff
@@ -747,9 +760,13 @@ def print_run_info(r):
             metrics: {'m': 1.5}
             status: FINISHED
         """
-        self._tracking_client.log_metric(run_id, key, value, timestamp, step)
+        return self._tracking_client.log_metric(
+            run_id, key, value, timestamp, step, synchronous=synchronous
+        )
 
-    def log_param(self, run_id: str, key: str, value: Any) -> Any:
+    def log_param(
+        self, run_id: str, key: str, value: Any, synchronous: Optional[bool] = True
+    ) -> Union[RunOperations, ParamValue]:
```

Comment:
Is it correct that could return a `ParamValue`? from the docstring it is either `RunOperations` or None.

Reply:
@chenmoneygithub I changed this based on this comment from @dbczumar , may be did not get it right based on the placement of the things. Let me know.

![image](https://github.com/mlflow/mlflow/assets/109864/031df9d6-af15-4ff5-9ee6-99902254d23a)

Reply:
Talked on the slack, making this wrapper inline in each file.